### PR TITLE
HPCC-30543 Rename WU Analyser to Cost Optimizer

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by shamser** [comment]
Documentation jira: https://track.hpccsystems.com/browse/HPCC-30655

File: tools/wutool/wutool.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ void usage(const char * action = nullptr)
                "   results <workunits> - Dump results from specified workunits\n"
                "   info <workunits> <filter>\n"
                "                       - Display information from a workunit\n"
-               "   analyze <workunit>  - Analyse the workunit to highlight performance issues\n"
+               "   analyze <workunit>  - Analyse the workunit to highlight potential cost saving\n"
```
Comment (jakesmith) [inline tools/wutool/wutool.cpp:91]: savings?

**Review by jakesmith — COMMENTED** [review]
The JIRA mentions:
>In WsWorkunits/WUInfo - but maybe in other places as well.

is it in WUInfo via common/wuanalysis code only?

Is there other places that need updating, e.g. documentation?

**Comment by shamser** [comment]
> tools/wutool/wutool.cpp

The wutool option is "analyze".  It is "Workunit Analyzer" or similar name is not referred to anywhere.

I have created a documentation jira (see above https://github.com/hpcc-systems/HPCC-Platform/pull/17954#issuecomment-1779601411).  However, it may be better to wait for the other analyzer improvement/changes to be completed before producing the documentation all these changes.

**Review by jakesmith — COMMENTED** [review]
@shamser - looks good, please squash

**Review by jakesmith — APPROVED** [review]

**Pull request merged**